### DISCUSSION TITLE
Concurrent  READ_ONLY with one READ_WRITE

### DISCUSSION THREAD

**Discussion by theblair** [duckdb/duckdb#14676] [Ideas]
Opening the same database file from multiple processes would be advantageous. Allowing multiple READ_ONLY connections with one READ_WRITE would be amazing. This should be possible with duckdb's MVCC approach to transactions without complex lock management. This would help with workloads that are read-mostly.

**Comment by Tishj** [comment]
Please see <https://duckdb.org/docs/connect/concurrency.html#handling-concurrency>

**Reply by theblair** [reply]
Thanks. Aware of this which is why I suggested the above improvement.

**Comment by tonne1** [comment]
Thought so too when I saw that duckdb uses MVCC -  the current conditions for concurrent operation seem unnecessarily restrictive. Supporting 1 writer - n readers concurrently would make duckdb suitable for a whole lot of broader usage scenarios.

**Comment by StephenFlavin** [comment]
When I saw that duckdb used MVCC I had incorrectly assumed that it would be good for concurrent access, in my scenario I've got infrequent writes and frequent reads on an in memory dB, when I apply a concurrent read load performance tanks.
Hoping that this can change in future 🙏
